### ISSUE TITLE
Is there a way to get chruby to automatically do something when switching?

### ISSUE THREAD

**Issue by tkelley65** [postmodern/chruby#346]
I have a ruby 1.8 install where I must set RUBYLIB manually before any ruby programs will work ... is it possible to have chruby do this automatically when switching to ruby 1.8? I don't see a way ... it would be nice to be able to take arbitrary actions when switching rubies ...

**Comment by postmodern** [comment]
You would have to write your own post-switching commands system.

**Comment by bronson** [comment]
Or use one of the zillions that already exist...  Here are some notes from a year ago when I needed to solve this.  (I ended up writing a one-off script, eschew overengineering)

* [autoenv](https://github.com/kennethreitz/autoenv): Overrides cd, that won't go well.
* [direnv](https://github.com/zimbatm/direnv): Well that's an awful lot of Go code.
* [anyenv](https://github.com/nikolay/anyenv): No README, looks kinda loose on security.
* [dotenv](https://github.com/bkeepers/dotenv): Ruby/Rails-specific, requires changing your source code.
* [chenv](https://github.com/betamatt/chenv): A lot of code plus a git-style UI.
  It's aopular name: [chenv](http://nanonanonano.net/linux/chenv), [chenv](https://github.com/bogdanjuszczak/chenv), etc.
* [aam](https://github.com/jlogsdon/aam): Apparently it's now named tem.
* [tem](https://github.com/Tapjoy/tem): Must be solving a completely different problem?

**Comment by postmodern** [comment]
@bronson right, the real problem is Bash doesn't support something like `preexec_functions`. You have to either inject a command into `PROMPT_COMMAND` (only works in interactive mode) or use `trap DEBUG` (not inherited by sub-processes). So most of these Bash solutions clobber any previously inserted Bash "hooks".

**Comment by ilikepi** [comment]
Would the idea proposed in #227 make this easier?

**Comment by postmodern** [comment]
@ilikepi providing a function would allow users to manually set the `trap DEBUG` command. Manually injection into `trap DEBUG` is still impossibly/difficult unless `set -T` is set.

**Comment by ilikepi** [comment]
Right... so does that mean you agree with this proposal (in a general sense)? I'm happy to put together a PR for further discussion.

**Comment by bitboxer** [comment]
`rbenv` has [plugins](https://github.com/rbenv/rbenv/wiki/Authoring-plugins) that enable you to do things when switching versions and other things. Would love to see that in chruby, too.
